tests/test_giturl.py

```diff
@@ -0,0 +1,59 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+from unittest import mock
+
+from fetchcode import giturl
+
+@mock.patch('os.system')
+def test_fetch_with_https_url_returns_a_response(mock_os):
+    url = 'https://github.com/TG1999/fetchcode.git'
+    with mock.patch('os.mkdir') as mocked_file:
+        response = giturl.fetch(source=url, branch='master', depth=None, dest='/home/tg')
+        assert response is not None
```

Comment:
Can you add to your tests? I would like to see what this `response` object contains for each of these url schemes.

Reply:
Add what @majurg I can not understand :sweat_smile:

Reply:
Well, you only test if `response` is not None. This is not very descriptive as to what fetching a git URL gets you. What are the attributes to this `response` objecTt?

Reply:
Okay I am getting you, doing the changes :)